60_networks.Rmd

```diff
@@ -0,0 +1,664 @@
+```
+
+
+```{r, eval=FALSE, echo=FALSE}
+saveRDS(peerj_net, file = "general/network_files/peerj_net.rds")
+```
+
+
+```{r, echo=FALSE}
+peerj_net <- readRDS("general/network_files/peerj_net.rds")
+```
+
+`netConstruct` returns an object of the class `microNet`, which contains all matrices generated during network construction. These are the basis for the network analysis.
+
+The object also contains an edge list, giving each edge's estimated association, dissimilarity, and adjacency. Let's take a quick look at the edges with the highest and lowest edge weights:
+
+```{r}
+edgelist <- peerj_net$edgelist1[order(peerj_net$edgelist1$adja, decreasing = TRUE), ]
+head(edgelist)
+tail(edgelist)
+```
+
+### Network analysis
+
+The `microNet` object is now passed to `netAnalyze` to perform network analysis.
+
+The function computes several common network analysis measures such as centrality measures, cluster assignment, the graphlet correlation matrix, as well as global network measures like the average path length and the clustering coefficient.
+
+Here, `centrLCC` is set to `TRUE` meaning that centralities are calculated only 
+for nodes in the largest connected component (LCC). 
+
+Clusters are identified using greedy modularity optimization 
+(by `cluster_fast_greedy` from [`igraph`](https://igraph.org/r/) package).
+
+Hubs are nodes with an eigenvector centrality value above the empirical 
+95% quantile of all eigenvector centralities in the network (argument `hubPar`).
+
+`weightDeg` and `normDeg` are set to `FALSE` so that the degree of a node is
+simply the number of its adjacent nodes.
+
+By default, a heatmap of the Graphlet Correlation Matrix (GCM) is returned (with 
+graphlet correlations in the upper triangle and significance codes resulting 
+from Student's t-test in the lower triangle). 
+See `?calcGCM` and `?testGCM` for details.
+
+```{r netAnalyze_single, fig.width=7, fig.height=7, out.width='75%'}
+peerj_netprops <- netAnalyze(peerj_net, 
+                             centrLCC = TRUE,
+                             clustMethod = "cluster_fast_greedy",
+                             hubPar = "eigenvector",
+                             hubQuant = 0.95,
+                             weightDeg = FALSE, 
+                             normDeg = FALSE)
+```
+
+```{r summary_single}
+summary(peerj_netprops)
+```
+
+Interpretation of some of the results:
+
+* The largest connected component (LCC) has 35 nodes and the network contains 6 singletons.
+* 10 clusters have been identified, containing two to eight nodes.
+* The detected hub nodes are Citrobacter, Escherichia, and Salmonella, which by definition are the nodes with the highest eigenvector centrality.
+* These three genera do also have a high degree and closeness centrality, but a lower betweenness centrality.
+* The average path length is 3.56. This means that on average it takes 3.56 steps (step length is the average dissimilarity) to get from one node to another node in the network.
```

Comment:
In case these numbers may change with random seed, consider automating the numbers in the text?

Reply:
Like instead of "35 nodes", it would be "`r nnodes` nodes" and the numbers automatically rendered